src/en/news/blog/2023/reef-freeze-rgw-performance/index.md

```diff
@@ -0,0 +1,179 @@
+
+# 4KB LIST
+
+![](images/Quincy-vs-Reef_-_4KB-List_-_200M-Total-Objects.svg)
+
+With more objects comes longer bucket list times vs the previous tests.  Reef is just slightly faster in the first iteration, but otherwise Quincy and Reef are neck and neck.
+
+# 4KB GET
+
+![](images/Quincy-vs-Reef_-_4KB-Get_-_200M-Total-Objects.svg)
+![](images/Quincy-vs-Reef_-_4KB-Get_-_200M_Total-Objects_-_Mako01-RGW.svg)
+![](images/Quincy-vs-Reef_-_4KB-Get_-_200M_Total-Objects_-_Mako01-OSD.svg)
+
+First, get performance is both a little faster and more stable in Reef than it was in Quincy.  Having said that, there are some really interesting behaviors here.  In Quincy, the first 2 minutes of testing are dominated by extremely high CPU usage.  Once that completes, the steady state CPU usage is quite a bit lower than it is in Reef.  OSD CPU usage is likewise lower.  Reef is using about 10-20% more CPU on the RGW side and about 17-23% more CPU on the OSD side in this test. This may ultimately be another regression that we need to bisect and track down.  The good news is that again, Reef's behavior appears to be much more consistent.  While RGW CPU usage is higher at the beginning of the test, it's no where near as high as what occured in the Quincy tests.
+
+# 4KB DELETE
+
+![](images/Quincy-vs-Reef_-_4KB-Delete_-_200M-Total-Objects.svg)
+
+Finally, Reef on average was about 5% faster at deleting 4K objects.  Again, this is an asyncronous process, so the results may not match how quickly RGW can delete things behind the scenes.
+
+# Conclusion
+
+These tests only show a small picture of RGW's performance characteristics.  We didn't have time for this article to look at single-op latencies, or try different client<->rgw topologies to minimize network traffic and push RGW and the OSDs harder.  There's many additional tests that could be run to give us more information.  Despite that, in the tests we did run, we saw two trends.  1) Reef showed anywhere from around 1-5% higher performance than Quincy, and it was fairly consistent across the tests and happened in most test iterations.  2) In some tests, especially the 4K GET tests, Reef showed higher CPU usage than Quincy.  This was more prevalant in the RGW processes but we also saw it in the OSD.  This is something we will likely want to follow up on before Reef is released.
```

Comment:
cc @mkogan1 @mattbenjamin 

you can also view the rendered blog post at https://wip-reef-freeze-rgw-performance.ceph.io/en/news/blog/2023/reef-freeze-rgw-performance/